docs/architecture-overview.rst

```diff
@@ -0,0 +1,200 @@
+Native iOS app <https://github.com/zulip/zulip-mobile>`__.
+
+We use `Transifex <https://www.transifex.com/zulip/zulip/>`__ to do
+translations.
+
+In this overview we'll mainly discuss the core Zulip web application.
+
+
+Usage assumptions and concepts
+==============================
+
+Zulip is a real-time web-based chat application meant for companies
+and similar groups ranging in size from 15 to more than a thousand
+users. It features real-time notifications, message persistence and
+search, public group conversations (*streams*), invite-only streams,
+private one-on-one and group conversations, inline image previews,
+team presence/a buddy list, a rich API, and several integrations with
+other services. The maintainer team aims to support users who connect
+to Zulip using dedicated iOS, Android, Linux, Windows, and Mac OS X
+clients, as well as people using modern web browsers or dedicated
+Zulip API clients.
+
+A server can host multiple Zulip *realms* (organizations) at the same
+domain, each of which is a private chamber with its own users,
+streams, customizations, and so on. This means that one person might
+be a user of multiple Zulip realms. The administrators of a realm can
+choose whether to allow anyone to register an account and join, or
+only allow people who have been invited, or restrict registrations to
+members of particular groups (using email domain names or corporate
+single-sign-on login for verification).
+
+
+Components
+==========
+
+
+Tornado and Django
+------------------
+
+Django is the main web application server; Tornado runs the
+server-to-client real-time push system. The app servers are configured
+by the Supervisor configuration (which explains how to start the
+server processes) and the nginx configuration (which explains which
+HTTP requests get sent to which app server).
+
+Tornado is an asynchronous server, so it can scale really well, but
+it's a real pain to write code for it. So Tornado gets only the parts
+that really need to scale because they involve maintaining a
+persistent connection from every single user. The parts that are
+activate relatively rarely (e.g. when people are typing) go into
+Django.
+
+
+nginx
+-----
+
+nginx is the front-end to all Zulip activity; it serves static assets
+and proxies to Django and Tornado. It handles HTTP requests according
+to the rules laid down in the many config files found in
+``zulip/puppet/zulip/files/nginx/``.
+
+``zulip/puppet/zulip/files/nginx/zulip-include-frontend/app`` is the
+most important of these files. It explains what happens when requests
+come in from outside.
+
+- All requests to URLs beginning with ``/static/`` are served from the
+  corresponding files in ``/home/zulip/prod-static/``.
+
+- Requests to ``/json/get_events``, ``/api/v1/events``, and
+  ``/sockjs`` are sent to the Tornado server running at
+  ``http://tornado``. These are requests to the real-time push system,
```

Comment:
http://tornado isn't it's actual URL -- nginx gets the hostname via puppet/zulip/files/nginx/zulip-include-frontend/upstreams

Note that websocket requests (currently used only to improve latency when **sending** messges) also go through Tornado.